server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/SetMessagesCreationProcessorTest.java

```diff
@@ -129,69 +144,42 @@ public void processShouldReturnNonEmptyCreatedWhenRequestHasNonEmptyCreate() thr
     @Test(expected = MailboxRoleNotFoundException.class)
     public void processShouldThrowWhenOutboxNotFound() {
         // Given
-        SetMessagesCreationProcessor<TestId> sut = new SetMessagesCreationProcessor<TestId>(null, null, null, null, null, null) {
-            @Override
-            protected Optional<Mailbox<TestId>> getOutbox(MailboxSession session) throws MailboxException {
-                return Optional.empty();
-            }
-        };
+        TestSystemMailboxesProvider doNotProvideOutbox = TestSystemMailboxesProvider.fakeOutbox(() -> Optional.empty());
+        SetMessagesCreationProcessor<TestId> sut = new SetMessagesCreationProcessor<>(null, null, null, null, doNotProvideOutbox);
         // When
         sut.process(buildFakeCreationRequest(), null);
     }
 
     @Test
-    @SuppressWarnings("unchecked")
     public void processShouldCallMessageMapperWhenRequestHasNonEmptyCreate() throws MailboxException {
         // Given
-        Mailbox<TestId> fakeOutbox = mock(Mailbox.class);
-        MessageMapper<TestId> mockMapper = mock(MessageMapper.class);
-        MailboxSessionMapperFactory<TestId> stubSessionMapperFactory = mock(MailboxSessionMapperFactory.class);
-        when(stubSessionMapperFactory.createMessageMapper(any(MailboxSession.class)))
-                .thenReturn(mockMapper);
-        MailSpool mockedMailSpool = mock(MailSpool.class);
-        MailFactory<TestId> mockedMailFactory = mock(MailFactory.class);
+        Mailbox<TestId> fakeOutbox = buildAndConfigureFakeMailbox("outbox", OUTBOX_ID);
 
-        SetMessagesCreationProcessor<TestId> sut = new SetMessagesCreationProcessor<TestId>(null, null,
-                stubSessionMapperFactory, new MIMEMessageConverter(), mockedMailSpool, mockedMailFactory) {
-            @Override
-            protected Optional<Mailbox<TestId>> getOutbox(MailboxSession session) throws MailboxException {
-                TestId stubMailboxId = mock(TestId.class);
-                when(stubMailboxId.serialize()).thenReturn("user|outbox|12345");
-                when(fakeOutbox.getMailboxId()).thenReturn(stubMailboxId);
-                when(fakeOutbox.getName()).thenReturn("outbox");
-                return Optional.of(fakeOutbox);
-            }
-        };
+        TestSystemMailboxesProvider stubbedMailboxes = TestSystemMailboxesProvider.fakeOutbox(() -> Optional.of(fakeOutbox));
+        SetMessagesCreationProcessor<TestId> sut = new SetMessagesCreationProcessor<>(
+                stubSessionMapperFactory, new MIMEMessageConverter(), mockedMailSpool, mockedMailFactory, stubbedMailboxes);
         // When
         sut.process(buildFakeCreationRequest(), buildStubbedSession());
 
         // Then
+        //noinspection unchecked
```

Comment:
this is a way to suppress reported compilation/style warnings for a given statement :
add's 2nd parameter is expecting a generic but is provided with a non generic one (any(MailboxMessage.class))
I can remove the line